artifactory/commands/transferfiles/transfer_test.go

```diff
@@ -0,0 +1,162 @@
+package transferfiles
+
+import (
+	"encoding/json"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/commands/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/transfer"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+)
+
+const uuidTokenForTest = "af14706e-e0c1-4b7d-8791-6a18bd1fd339"
+
+type validateVersionTestSuite struct {
+	testName      string
+	curVersion    string
+	errorExpected bool
+}
+
+func TestValidateDataTransferPluginMinimumVersion(t *testing.T) {
+	var pluginVersion string
+	testServer, serverDetails, _ := transfer.CreateRestsMockServer(t, func(w http.ResponseWriter, r *http.Request) {
+		if r.RequestURI == "/"+pluginsExecuteRestApi+"dataTransferVersion" {
+			content, err := json.Marshal(utils.VersionResponse{Version: pluginVersion})
+			assert.NoError(t, err)
+			_, err = w.Write(content)
+			assert.NoError(t, err)
+		}
+	})
+	defer testServer.Close()
+
+	srcPluginManager, err := createSrcRtUserPluginServiceManager(serverDetails)
+	if err != nil {
+		assert.NoError(t, err)
+		return
+	}
+
+	testsArray := []validateVersionTestSuite{
+		{"valid version", "9.9.9", false},
+		{"exact version", dataTransferPluginMinVersion, false},
+		{"invalid version", "1.0.0", true},
+	}
+
+	for _, test := range testsArray {
+		t.Run(test.testName, func(t *testing.T) {
+			pluginVersion = test.curVersion
+			err = getAndValidateDataTransferPlugin(srcPluginManager)
+			if test.errorExpected {
+				assert.EqualError(t, err, getMinimalVersionErrorMsg(test.curVersion))
+				return
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestVerifyConfigImportPluginNotInstalled(t *testing.T) {
+	testServer, serverDetails, _ := transfer.CreateRestsMockServer(t, func(w http.ResponseWriter, r *http.Request) {
+		if r.RequestURI == "/"+pluginsExecuteRestApi+"dataTransferVersion" {
+			w.WriteHeader(http.StatusNotFound)
+			_, _ = w.Write([]byte("Not found"))
+		}
+	})
+	defer testServer.Close()
+
+	srcPluginManager, err := createSrcRtUserPluginServiceManager(serverDetails)
+	if err != nil {
+		assert.NoError(t, err)
+		return
+	}
+
+	_, err = srcPluginManager.version()
+	assert.ErrorContains(t, err, "Response from Artifactory: 404 Not Found.")
+}
+
+func TestUploadChunkAndPollUploads(t *testing.T) {
```

Comment:
This function is way too big and 
Let's break it into more functions to make it more readable.